### PR TITLE
[DOCS] Adds complete Cohere tutorial

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

[Open In Colab](https://colab.research.google.com/drive/1B014U7--4PDRgUNT5sRseUCiNHk0lpSx)

This PR adds a tutorial to the ES docs set that demonstrates the full capabilities of the Cohere service in Elasticsearch.
It is similar to and inspired by [this Cohere page](https://docs.cohere.com/docs/elasticsearch-and-cohere) but uses a different data set to give users more working examples.

### Preview on docs site

[Cohere tutorial](https://elasticsearch_bk_108415.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/cohere-es.html)

### CODE REVIEW

File: docs/reference/search/search-your-data/cohere-es.asciidoc [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+<titleabbrev>Using Cohere with {es}</titleabbrev>
+++++
+
+The instructions in this tutorial shows you how to compute embeddings with
+Cohere using the {infer} API and store them for efficient vector or hybrid
+search in {es}. This tutorial will use the Python {es} client to perform the
+operations.
+
+You'll learn how to:
+* create an {infer} endpoint for text embedding using the Cohere service,
```
Comment (jonathan-buttner) [inline docs/reference/search/search-your-data/cohere-es.asciidoc:13]: Should the `*` render in a specific way?

Looks like this in my browser when I view the preview

image

File: docs/reference/search/search-your-data/cohere-es.asciidoc [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+    },
+)
+------------------------------------------------------------
+
+
+[discrete]
+[[cohere-es-infer-pipeline]]
+==== Create the {infer} pipeline
+
+You have an {infer} endpoint and an index ready to store embeddings. The next
```
Comment (jonathan-buttner) [inline docs/reference/search/search-your-data/cohere-es.asciidoc:144]: nit: Maybe this should start with `Now you have...`?

File: docs/reference/search/search-your-data/cohere-es.asciidoc [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+The response will look similar to this:
+
+[source,consol-result]
+--------------------------------------------------
+Query: What is biosimilarity?
+Response: Biosimilarity is based on the comparability concept, which has been used successfully for several decades to ensure close similarity of a biological product before and after a manufacturing change. Over the last 10 years, experience with biosimilars has shown that even complex biotechnology-derived proteins can be copied successfully.
+Sources:
+Interchangeability of Biosimilars: A European Perspective: (...)
+--------------------------------------------------
+// NOTCONSOLE
```
Comment (jonathan-buttner) [inline docs/reference/search/search-your-data/cohere-es.asciidoc:370]: I wonder if it'd be helpful to have the full python code combined together in a python script and have it checked into elasticsearch somewhere. I've found that helpful when going through other tutorials and I need to jump to the code.

**Review by jonathan-buttner — APPROVED** [review]

File: docs/reference/search/search-your-data/cohere-es.asciidoc (hunk 3 above)
Comment (szabosteve) [reply docs/reference/search/search-your-data/cohere-es.asciidoc:370]: @jonathan-buttner I don't know how to do that (I mean the part about "checking into ES somewhere") but I'm happy to learn. I certainly can create a `.py` file with all the code.

**Pull request merged**